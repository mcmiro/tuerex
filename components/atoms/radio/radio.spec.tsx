import * as React from 'react';
import { screen, render, fireEvent } from '@testing-library/react';
import Radio from './';

describe('Test Radio component', () => {
  it('Should pass when name added', () => {
    render(<Radio name="test" />);

    const radio = screen.getByRole('radio');
    expect(radio).toBeDefined();
    expect(radio).toMatchSnapshot();
  });

  it('Should pass when radio is not checked', () => {
    const changeMock = jest.fn();
    render(<Radio name="test" onChange={changeMock} />);

    const radio = screen.getByRole('radio') as HTMLInputElement;
    expect(radio.checked).toBe(false);
  });

  it('Should pass when component click radio and change checked to true', async () => {
    const changeMock = jest.fn();
    render(<Radio name="test" onChange={changeMock} />);

    const radio = screen.getByRole('radio') as HTMLInputElement;
    fireEvent.click(radio);
    expect(changeMock).toHaveBeenCalledTimes(1);
    expect(radio).toBeChecked();
  });
});
