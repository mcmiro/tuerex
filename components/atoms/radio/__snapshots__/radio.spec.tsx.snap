// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Radio component Should pass when name added 1`] = `
<input
  class="form-radio w-6 h-6 ring-1 ring-black text-black bg-white disabled:ring-gray-500 disabled:text-gray-500 disabled:event-pointer-none"
  id=":r0:"
  name="test"
  type="radio"
  value=""
/>
`;
